src/components/organisms/chat/active_room.rs

```diff
@@ -103,46 +109,129 @@ pub fn ActiveRoom(cx: Scope) -> Element {
         });
     };
 
-    let on_handle_attach = move |event: Attachment| {
-        send_attach.send(event);
+    let on_handle_attach = move |attachment: Attachment, send_to_thread: bool| {
+        send_attach.send(AttachmentStream {
+            attachment,
+            send_to_thread,
+        });
     };
 
-    info!("{:?} {:?}", room.get(), *current_room.read());
-
     cx.render(rsx! {
-        div {
-            style: "
-                display: flex;
-                flex-direction: column;
-                height: 100vh;
-            ",
-            Header {
-                text: "{current_room.read().name.clone()}",
-                avatar_element: render!(rsx!(
-                    Avatar {
-                        name: (*current_room.read()).name.to_string(),
-                        size: 32,
-                        uri: current_room.read().avatar_uri.clone()
+            // Room messages
+            div {
+                class: "active-room",
+                Header {
+                    text: "{current_room.read().name.clone()}",
+                    avatar_element: render!(rsx!(
+                        Avatar {
+                            name: (*current_room.read()).name.to_string(),
+                            size: 32,
+                            uri: current_room.read().avatar_uri.clone()
+                        }
+                    )),
+                    on_event: header_event
+                }
+                List {
+                    messages: messages.clone(),
+                    thread: None,
+                    is_loading: is_loading,
+                    on_scroll: move |_| {
+                        use_m.loadmore(current_room.read().id.clone());
                     }
-                )),
-                on_event: header_event
-            }
-            List {},
-            InputMessage {
-                message_type: InputType::Message,
-                placeholder: input_placeholder.get().as_str(),
-                on_submit: on_push_message,
-                on_event: input_message_event,
-                on_attach: on_handle_attach
+                },
+                InputMessage {
+                    message_type: InputType::Message,
+                    placeholder: input_placeholder.get().as_str(),
+                    on_submit: move |event| {
+                        on_push_message(event, false)
+                    },
+                    on_event: input_message_event,
+                    on_attach: move |event|{
+                        on_handle_attach(event, false)
+                    }
+                }
             }
-        }
 
+            if let Some(t) = timeline_thread.read().deref() {
+                let head_message = &t.thread[t.thread.len() - 1];
+                let x = &head_message.body;
+
+                let title_thread = match x {
+                    TimelineMessageType::Image(file) => {
+                        String::from(file.body.clone())
```

Comment:
looks redundant to have `String::from` and `clone` either one of the should be enough